src/plugins/intel_cpu/src/dnnl_postops_composer.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "dnnl_postops_composer.h"
+
+#include <common/primitive_attr.hpp>
+
+#include "node.h"
+
+namespace ov {
+namespace intel_cpu {
+
+DnnlPostOpsComposer::DnnlPostOpsComposer(ov::intel_cpu::Node* node,
+                                         dnnl::primitive_attr& attr,
+                                         dnnl::post_ops& ops,
+                                         std::vector<MemoryPtr>& args,
+                                         const VectorDims& outputDims,
+                                         int indexOfOutputChannelDim,
+                                         bool isINT8)
+    : node(node),
+      attr(attr),
+      ops(ops),
+      args(args),
+      outputDims(outputDims),
+      idxOC(indexOfOutputChannelDim),
+      isINT8(isINT8) {
+    IE_ASSERT(idxOC >= 0 && idxOC < outputDims.size());
+    OC = outputDims[idxOC];
+    dimsPerOC = dimsPerTensor = VectorDims(outputDims.size(), 1);
+    dimsPerOC[idxOC] = OC;
+    oscale_mask = 0;
+    oscale_values = {1.0f};
+}
+
+DnnlPostOpsComposer::~DnnlPostOpsComposer() {
```

Comment:
Seems to be not safe. If smb will decide to create DnnlPostOpsComposer object as cpu node field destructor won't be called before attr/args usage.

Reply:
Yes, agree, I changed the design so each time output scale is changed, the dynamic argument corresponding to it gets updated immediately,  in that way there is no need to to it at the end of whole setPostOps process. it will be more safe.

but I have to keep a fixed entry in the `postOpsArgs` container, for I have to update the same entry each time output scale is changed, and I feel it's very inconvenient to do it with `postOpsArgs` being the `vector` typed container, so I changed it into `unordered_map` based container so I can locate that entry explicitly and also it's more align with the container type used in oneDNN for dynamic arguments.

Please note this change effects one overload of legacy API `appendPostOps` also